src/bin/pg_autoctl/cli_get_set_properties.c

```diff
@@ -668,22 +668,27 @@ static bool
 set_node_candidate_priority(Keeper *keeper, int candidatePriority)
 {
 	KeeperConfig *config = &(keeper->config);
-	NodeAddressArray nodesArray = { 0 };
+	CurrentNodeStateArray nodesArray = { 0 };
 
 	/*
 	 * There might be some race conditions here, but it's all to be
 	 * user-friendly so in the worst case we're going to be less friendly that
 	 * we could have.
 	 */
-	if (!monitor_get_nodes(&(keeper->monitor),
-						   config->formation,
-						   config->groupId,
-						   &nodesArray))
-	{
-		/* ignore the error, just don't wait in that case */
-		log_warn("Failed to get_nodes() on the monitor");
+	if (!monitor_get_current_state(&(keeper->monitor),
+								   config->formation,
+								   config->groupId,
+								   &nodesArray))
+	{
+		log_fatal("Failed to get the list of all the nodes in formation \"%s\" "
+				  "from the monitor, see above for details",
+				  keeper->config.formation);
+		return false;
```

Comment:
We now do a hard failure here. Is that what we want? (see old comment about ignoring the error and just not waiting in that case).

Reply:
Good point, fixed now!